src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -110,17 +116,103 @@ CAC.Map.Control = (function ($, L, _) {
     }
 
     /**
-     * Plot an array of geojson points onto the map
+     * Add isochrone outline to map
+     */
+    function drawIsochrone(isochrone) {
+        isochroneLayer = L.geoJson(isochrone, {
+            style: {
+                'color': 'red',
+                'opacity': 0.8
+            }
+        }).addTo(map);
+    }
+
+    /**
+     * Fetch all the reachable destinations within our destination database,
+     * and their enclosing isochrone (travelshed).
+     *
+     * @return {Object} Promise resolving to JSON with 'matched' and 'isochrone' properties
+     */
+    function fetchReachable(payload) {
+        var isochroneUrl = '/map/reachable';
+        var deferred = $.Deferred();
+        if (payload.coords && payload.mode && payload.date &&
+            payload.time && payload.maxTravelTime && payload.maxWalkDistance) {
+            $.ajax({
+                type: 'GET',
+                data: payload,
+                url: isochroneUrl,
+                contentType: 'application/json'
+            }).then(deferred.resolve);
+        } else {
+            deferred.fail();
+        }
+        return deferred.promise();
+    }
+
+    /**
+     * Get travelshed and destinations within it, then display results on map.
+    */
+    function fetchIsochrone() {
+        // clear results of last search
+        clearDiscoverPlaces();
+
+        var getIsochrone = function(params) {
+            fetchReachable(params).then(function(data) {
+                drawIsochrone(data.isochrone);
+                // also draw 'matched' list of locations
+                var matched = _.pluck(data.matched, 'point');
+                drawDestinations(matched);
+            }, function(error) {
+                console.error(error);
+            });
+        };
+
+        // default to current date and time
+        var now = new Date();
+        // months are zero-based
+        var dateStr = [(now.getMonth() + 1),
+                        now.getDate(),
+                        now.getFullYear()
+                      ].join('-');
```

Comment:
Yes, I'm thinking this will get moved and significantly modified when we implement user preferences and options.  Right now, there's no way to set the time for any searches, which I think will be pretty important.